ranger-zk-server/src/main/java/com/flipkart/ranger/server/manager/RangerBundleManager.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2015 Flipkart Internet Pvt. Ltd.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.flipkart.ranger.server.manager;
+
+import com.flipkart.ranger.server.bundle.RangerServerBundle;
+import io.dropwizard.lifecycle.Managed;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.curator.framework.CuratorFramework;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+@Getter
+@Slf4j
+public class RangerBundleManager implements Managed {
```

Comment:
We could have a dedicated manager to start stop the `RangerServerBundle` as part of ranger-service-bundle (removing RangerHttpBundleManager), and delegate the `CurationFramework` management separately to the ranger-zk-server package.<br>
Either way is fine actually, but i felt that if someone were to use ranger-service-bundle in his dw-application, he would have to remember to explicitly do the RangerServerBundle management.

Reply:
Can do this, but the curatorFramework needs to be started before the bundle manager is started and if both of them are inside different lifecycle scopes, there may be no determinism on what starts first, so contained them both in a single scope in the zk package. So had to create the http manager separately. :)